test/android/util/runtime_client/src/org/xwalk/test/util/XWalkRuntimeClientTestGeneric.java

```diff
@@ -17,12 +20,20 @@
         extends ActivityInstrumentationTestCase2<T> {
     private XWalkRuntimeView mRuntimeView;
     XWalkRuntimeClientTestUtilBase mTestUtil;
+    private Timer mTimer = new Timer();
 
     @Override
     protected void setUp() throws Exception {
         super.setUp();
+        waitForXWalkReady();
```

Comment:
setUp runs in the instrumentation test thread, some initializations for test thread needs to wait for XWalk ready. In the newest code, AsyncTask was used for XWalk initialization.
Test case could be run in instrumentation test thread and also in UI thread with ```UiThreadTest```.